spec/30-processing-model.md

```diff
@@ -0,0 +1,69 @@
+   - If the version prefix cannot be parsed, the tracer creates   a new `traceparent` header and deletes `tracestate`.
+   - If the version number is higher than supported by the tracer, the
+     implementation uses the format defined in this specification to parse
+     `trace-id` and `parent-id`. The tracer will only parse `trace-flags` values
+     supported by the current version of this specification and ignore all other
+     values. If parsing fails, the tracing system creates a new `traceparent`
+     header and deletes the `tracestate`.
+
+3. If the tracer supports the version number, it validates `trace-id` and
+   `parent-id`.
+   - If either `trace-id`, `parent-id` or `trace-flags`  are invalid,  the tracer
+     creates a new `traceparent` header and deletes `tracestate`.
+
+4. The tracer validates the `tracestate` header. If the `tracestate` header
+   cannot be parsed the tracer deletes it.
+
+5. For each outgoing request the tracer performs the following steps:
+
+   - The tracing system MUST modify the `traceparent` header.
+        - **Update `parent-id`**. The value of property `parent-id` MUST be set
+          to a value representing the ID of the current operation.
+        -  **Update `recorded`**. The value of `recorded` reflects the caller's
+            recording behavior. The value of the `recorded` flag of
+            `trace-flags` MAY be set to `1` if the trace data is likely to be
+            recorded or to `0` otherwise. Setting the flag is no guarantee that
+            the trace will be recorded but increases the likeliness of
+            end-to-end recorded traces.
+    - The tracer MAY modify the `tracestate` header
+        - **Update key value**. The value of any key can be updated. Modified
+          keys MUST be moved to the beginning of the list.
+        - **Add new key-value pair**. The new key-value pair should be added
+          into the beginning of the list.
+        - **Delete the key-value pair**. Any key-value pair MAY be deleted. It
+          is highly discouraged to delete keys that weren't generated by the
+          same tracing system or platform. Deletion of an unknown key-value pair
+          will break correlation in other systems. This mutation enables two
+          scenarios:
+            - Proxies can block specific `tracestate` keys for privacy and
+              security concerns.
+            - Truncation of long `tracestate` entries.
+    - The tracer set the `traceparent` and `tracestate` header for the outgoing
+      request
+
+
+## Alternative Processing
+
+The processing model above describes the complete set of steps for processing
+Trace Context headers. There are, however, situations when an implementation
+might only support a subset of the steps described above. Proxies or messaging
+middleware MAY decide to not modify the `traceparent` headers but may remove invalid
```

Comment:
uppercased "MAY"? Or just no "may" at all as it is an example that already started with MAY.

```suggestion
middleware MAY decide to not modify the `traceparent` headers but remove invalid
```